tests/helpers.rs

```diff
@@ -31,132 +33,74 @@ fn get_random_tcp_port() -> u16 {
         .port()
 }
 
-#[derive(Debug)]
-pub struct CleanupGuard {
-    child: Option<Child>,
-
-    already_running: bool,
+pub struct TestEnv {
+    nats_server: ContainerAsync<GenericImage>,
+    wasmcloud_hosts: Vec<ContainerAsync<GenericImage>>,
 }
 
-impl Drop for CleanupGuard {
-    fn drop(&mut self) {
-        if !self.already_running {
-            match std::process::Command::new("wash").args(["down"]).output() {
-                Ok(o) if !o.status.success() => {
-                    eprintln!(
-                        "Error stopping wasmcloud host: {}",
-                        String::from_utf8_lossy(&o.stderr)
-                    )
-                }
-                Err(e) => eprintln!("Error stopping wasmcloud host: {e}"),
-                _ => (),
-            }
-        }
+impl TestEnv {
+    pub async fn nats_port(&self) -> u16 {
+        self.nats_server
+            .get_host_port_ipv4(DEFAULT_NATS_PORT)
+            .await
+            .unwrap()
     }
-}
-
-/// Configuration struct for wash instances that are used for testing
-#[derive(Debug, Default)]
-pub struct TestWashConfig {
-    /// Port on which to run wasmCloud
-    pub nats_port: Option<u16>,
 
-    /// Only connect to pre-existing NATS instance
-    pub nats_connect_only: bool,
-}
-
-impl TestWashConfig {
-    /// Build a test wash configuration with randomized ports
-    pub async fn random() -> Result<TestWashConfig> {
-        let nats_port = Some(get_random_tcp_port());
-
-        Ok(TestWashConfig {
-            nats_port,
-            ..TestWashConfig::default()
-        })
+    pub async fn nats_url(&self) -> String {
+        let nats_host = self.nats_server.get_host().await.unwrap();
+        let nats_port = self.nats_port().await;
+        format!("{nats_host}:{nats_port}")
     }
 
-    /// Get the NATS URL for this config
-    pub fn nats_url(&self) -> String {
-        format!("127.0.0.1:{}", self.nats_port.unwrap_or(DEFAULT_NATS_PORT))
+    pub async fn nats_client(&self) -> async_nats::Client {
+        async_nats::connect(self.nats_url().await).await.unwrap()
```

Comment:
I think this would do well as a result that we return, since the connection is more likely to fail than other things

Reply:
Same re: returning the error to the caller and handling it closer to the code that's calling in.